src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -101,30 +88,36 @@ public AsyncJsonTokenParser(
    * or equivalent constructor of {@link JsonToken} subclasses.
    */
   public ListenableFuture<JsonToken> verifyAndDeserialize(String tokenString) {
-    ListenableFuture<String> futureTokenString = Futures.immediateFuture(tokenString);
-    AsyncFunction<String, JsonToken> verifyFunction =
-        tokenStringInput -> {
-          JsonToken jsonToken = deserialize(tokenStringInput);
-          Function<Void, JsonToken> returnFunction = unused -> jsonToken;
-          return Futures.transform(verify(jsonToken), returnFunction, executor);
-        };
+    JsonToken token;
+    try {
+      token = deserialize(tokenString);
+    } catch (Throwable t) {
+      return Futures.immediateFailedFuture(t);
+    }
+
+    ListenableFuture<JsonToken> futureTokenString = Futures.immediateFuture(token);
```

Comment:
Do you still need the futureTokenString or transformAsync? I think you can just call verify(jsonToken) directly, and transform that future to return the jsonToken.

Reply:
Yeah, we don't need futureTokenString anymore. It has been removed altogether.